File: cpp/48_旋转图像_medium.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int N = matrix.size();
        auto tmp = matrix;
        for(int i=0; i<N; i++)
            for(int j=0; j<N; j++)
                matrix[j][N-i-1] = tmp[i][j];
    }
};

File: python/48_旋转图像_medium.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        N = len(matrix[0])
        tmp = []
        for i in range(N):
            tmp.append(matrix[i][:])
        for i in range(N):
            for j in range(N):
                matrix[j][N-i-1] = tmp[i][j]
        return matrix